tools/query_breakdown/src/main/java/com/google/bigquery/Node.java

```diff
@@ -27,15 +27,91 @@
   private int unparseableCount;
 
   // indicates how we handled the error
-  enum errorHandlingType {
+  enum ErrorHandlingType {
     DELETION,
     REPLACEMENT
   }
 
+  ErrorHandlingType type;
+
+  /**
+   * constructor for the node class for deletion
+   */
+  public Node (Node parent, int startLine, int startColumn, int endLine, int endColumn,
+      int unparseableCount) {
+    this.parent = parent;
+    this.startLine = startLine;
+    this.startColumn = startColumn;
+    this.endLine = endLine;
+    this.endColumn = endColumn;
+    type = ErrorHandlingType.DELETION;
+    this.unparseableCount = unparseableCount;
+  }
+
   /**
-   * constructor for the node class
+   * constructor for the node class for replacement
    */
   public Node (Node parent, int startLine, int startColumn, int endLine, int endColumn,
-      boolean errorHandlingType, String replaceFrom, String replaceTo, int unparseableCount) {
+      String replaceFrom, String replaceTo, int unparseableCount) {
+    this.parent = parent;
+    this.startLine = startLine;
+    this.startColumn = startColumn;
+    this.endLine = endLine;
+    this.endColumn = endColumn;
+    type = ErrorHandlingType.REPLACEMENT;
+    this.replaceFrom = replaceFrom;
+    this.replaceTo = replaceTo;
+    this.unparseableCount = unparseableCount;
+  }
+
+  /**
+   * Override toString method for better debugging
+   */
+  @Override
+  public String toString() {
+    return "Parent: " + parent.getStartColumn() + parent.getEndColumn()
```

Comment:
Nit: try to use str.format over + from the readability standpoint.

Reply:
Got it, changed it for the node class as well as the querybreakdown class!